doc/adr/0007-store-names-sentence-dates-and-prison-names-in-our-database.md

```diff
@@ -0,0 +1,92 @@
+  Redis, despite it being a more short-lived approach.
+- The Prisoner Search service only allows for searching for 1000 prison numbers at once (when fetching the data to
+  cache, we'd be making a large bulk request for multiple Prison Numbers at once), so as the service grows, we'd need to
+  find an approach for doing this efficiently.
+
+## Decision
+
+We will not use caching to store the fields from external services, but will instead store these values in our own
+Postgres database to make use of JPA sorting.
+
+From the Prisoner Search API, we will fetch and store the following details about the referred person:
+
+- `prisonNumber`
+- `bookingId`
+- `firstName`
+- `lastName`
+- `indeterminateSentence`
+- `nonDtoReleaseDateType`
+- `conditionalReleaseDate`
+- `tarriffDate`
+- `paroleEligibilityDate`
+- `earliestReleaseDate` (one of the dates above, based on whether person is on an indeterminate sentence)
+
+From the Prison Register API, we will fetch and store the `name` of a Prison in our service.
+
+These fields will be fetched and stored on a Referral (linking to a new Person table) when the draft Referral is
+created.
+
+To prevent this data from getting out of date, e.g. someone's name or their sentence details changing we will initially
+re-fetch data from the Prisoner Search when a user clicks on the entry in their caselist.
+
+Longer term, however, we'll want to listen out for domain events emitted by
+the [Offender Events service](https://offender-events-ui-dev.prison.service.justice.gov.uk/messages?text-filter=) when a
+record is updated and update Referrals in our own database accordingly.
+
+When someone creates a new referral, we fetch and store their name and sentence dates (the columns we want to sort by)
+in our real database, rather than a cache. This means we can make use of the current sorting functionality and wouldn't
+have to hand-write everything ourselves.
+
+## Consequences
+
+### Benefits
+
+Storing this data in our database will greatly simplify our code in contrast to the complexity introduced by
+implementing caching, and we'll be able to make use of the existing filtering and sorting functionality provided by JPA.
+
+It will also speed up parts of the service where we previously would have made requests to other services for this
+information on read. We will instead be able to read these values from our own database and skip this round trip.
+
+### Disadvantages
+
+We'll be storing extra details about a referred person (their name and sentence dates) in our own database. This is
+something we've actively been trying to avoid as much as possible, preferring to read this directly from other sources
+to prevent storing any more personal data in our service than is necessary. However, the importance of sorting by these
+fields has made it necessary for us to store this data.
+
+We'll need to ensure we keep this data up to date and inform users of the time and date the data was fetched to ensure
```

Comment:
I think we might explore the option of leaving the data as-is, at the 'point of referral' and highlight any records that may have changed in the interim, we're discussing testing this prior to R1, so will update once we've floated the idea past users.

Reply:
I'll add this in to make sure it's captured as either a decision or consequence, thanks for the reminder.